### ISSUE TITLE
Rationalize validation and error handling

### ISSUE THREAD

**Issue by tiltec** [yunity/karrot-frontend#668]
groupEdit and storeEdit are already in quite a good shape, need to bring the other forms there too.

Add client-side validation (vuelidate)
- [ ] pickupEdit
- [ ] pickupSeriesEdit
- [x] agreementEdit
- [ ] login
- [ ] passwordReset (check if email?)
- [ ] signup
- [ ] ProfileEdit
- [ ] Settings.vue (require display name...)
- [ ] changeEmail

Add server validation error handling
- [x] invitationsUI (does only nonFieldErrors right now)
- [ ] agreementEdit (quite complex, won't do it)
- [x] groupInfoCard (only shows "password wrong", without checking)
- [x] login (needs rework, see #658)
- [x] ProfileEdit
- [x] changePassword
- [x] changeEmail
- [x] WallInput
- [x] VerificationWarning
- [x] verifyEmail

Things to look out for:
- disable submit if request is already in progress
- add visual feedback if request is in progress (loader, spinner)

Other related tasks:
- [x] storeedit, groupedit need a spinner
- [x] pickup (series) edit need a `canSave` function refactored

**Comment by tiltec** [comment]
Also need to handle server permission errors, e.g. the server returns 403 with 

```json
{ "detail": "You are already a member." }
```

**Comment by tiltec** [comment]
Talked a bit about it with @nicksellen 

    two kinds of errors: validation (HTTP400-499) and other errors
    Validation errors are caught and handled inside the store as data
    Other errors are handled in interceptors
    Goals: other errors end up in Sentry, no validation errors end up in sentry
    
    DRF: field errors, nonFieldErrors (of validation errors), detail (of permission errors)
    
    Now we have three kinds:
        - validation errors - the user can handle them
        - permission errors - the user can maybe handle them, maybe not. Should probably not make the request in the first place (out-of-date information)
        - other errors
        
    Origin of errors:
        - our of date auth information
        - followed wrong link
        - programming error
        
    DRF HTTP errors:
        - 400: returns field or nonFieldError
        - 401, 403: returns detail

**Comment by tiltec** [comment]
#691 is related, most request handling happens in vuex stores

**Comment by tiltec** [comment]
We currently don't use the `detail` field of `401` and `403` responses at all.

**Comment by tiltec** [comment]
This is partially dependent on #626, it will introduce a nice and uniform way to access validation errors across all vuex modules.

**Comment by tiltec** [comment]
I keep getting confused about the different components and which stages they are in, so I made this overview. I intend to keep it updated until this issue is resolved.

Explanation:
`edit` means using objectDiff by including `editMixin`, `status` means handling of server-side validation by including `statusMixin`, `vuelidate` means client-side form validation.
Additionally, I list the places where deprecated patterns are used: `old-style objectDiff` should be replaced by `editMixin`, `old-style status` should be replaced by `statusMixin`

Replacing `old-style status` by `statusMixin` also means using the new `meta` module stuff, so it needs additional changes in the related vuex module.

- PickupEdit: edit, status,
- PickupSeriesEdit: edit, status,
- ProfileEdit: edit,
- AgreementForm: vuelidate, old-style objectDiff,
- GroupEdit: vuelidate, old-style objectDiff,
- InvitationsForm: vuelidate (needs checking #761),
- StoreEdit: vuelidate, old-style status, old-style objectDiff
- PasswordReset: old-style status,
- Signup: old-style status,

**Comment by tiltec** [comment]
I wrote myself a little checklist for going through a form:

- is it an edit form? -> use `editMixin`
- uses `statusMixin`?
    - uses hasError & firstError to check for server-side validation errors?
    - uses `hasNonFieldErrors` & `nonFieldErrors` somewhere?
    - sets `:loader="isPending"` on submit button?
    - does *not* prevent save if it has server-side validation errors?
    - uses pending status for non-submit async actions? (e.g. destroy)
- uses vuelidate ($v) to check for client-side validation error?
    - prevents save if it has client-side validation errors?

I'll add this to the docs at some point! Anything else?

**Comment by tiltec** [comment]
Password reset does nothing instead of showing error when email does not exist.

**Comment by tiltec** [comment]
login page does not show error messages: Property or method "firstValidationError" is not defined on the instance but referenced during render

**Comment by tiltec** [comment]
I think most of the mentioned tasks are solved by now. I decided against adding vuelidate to most forms, as it wouldn't improve UX much and it definitely adds complexity and translation load.

The only place where server-side validation errors are not shown is `agreementForm` because I found it too complex to do in a good way. As it is not _really_ used in production, I would leave it for later.
